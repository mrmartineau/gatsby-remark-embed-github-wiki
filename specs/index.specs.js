jest.mock("request-promise");

import remark from "remark";
import plugin from "../src/index";

const getNodeContent = node => node.children[0].children[0];

describe("gatsby-remark-embed-github-wiki", () => {
  it("generates html with username, repo and page", async () => {
    const markdownAST = remark.parse(
      "`gh-wiki:visitinc/visitinc.github.io/Index-Copy`"
    );

    const processed = await plugin({ markdownAST });
    expect(processed).toBeTruthy();

    expect(getNodeContent(markdownAST)).toMatchSnapshot();
  });

  it("generates html with default username and repo", async () => {
    const markdownAST = remark.parse("`gh-wiki:Index-Copy`");

    const processed = await plugin(
      { markdownAST },
      { username: "visitinc", repo: "visitinc.github.io" }
    );
    expect(processed).toBeTruthy();

    expect(getNodeContent(markdownAST)).toMatchSnapshot();
  });

  it("inline username and repo overrides configuration", async () => {
    const markdownAST = remark.parse(
      "`gh-wiki:visitinc/visitinc.github.io/Index-Copy`"
    );

    const processed = await plugin({ markdownAST }, { username: "john" });
    expect(processed).toBeTruthy();

    expect(getNodeContent(markdownAST)).toMatchSnapshot();
  });

  it("fails when no username or repo is provided", async () => {
    const markdownAST = remark.parse("`gh-wiki:Index-Copy`");
    expect(plugin({ markdownAST })).rejects.toEqual(
      new Error("Missing username or repo information")
    );
  });

  it("fails when no page is provided", async () => {
    const markdownAST = remark.parse("`gh-wiki:visitinc/visitinc.github.io`");
    expect(plugin({ markdownAST })).rejects.toEqual(
      new Error("Missing page information")
    );
  });

  it("ignores everything that is not inline code", async () => {
    const markdownAST = remark.parse("# Syntax");
    const originalMarkdownAST = markdownAST;

    const processed = await plugin({ markdownAST });
    expect(processed).toBeTruthy();

    expect(markdownAST).toBe(originalMarkdownAST);
  });
});
