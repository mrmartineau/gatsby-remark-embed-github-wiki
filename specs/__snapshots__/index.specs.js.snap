// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gatsby-remark-embed-github-wiki generates html with default username and repo 1`] = `
Object {
  "position": Position {
    "end": Object {
      "column": 21,
      "line": 1,
      "offset": 20,
    },
    "indent": Array [],
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "html",
  "value": "<p>Studio Visit App is:</p>
<ul>
<li>a slick but informal scheduling tool</li>
<li>a place to document and share your work</li>
<li>an index of practices searchable by location, medium, gender, allergy, discourse, and more</li>
<li>a federation of safe, user-moderated communities </li>
</ul>",
}
`;

exports[`gatsby-remark-embed-github-wiki generates html with username, repo and page 1`] = `
Object {
  "position": Position {
    "end": Object {
      "column": 49,
      "line": 1,
      "offset": 48,
    },
    "indent": Array [],
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "html",
  "value": "<p>Studio Visit App is:</p>
<ul>
<li>a slick but informal scheduling tool</li>
<li>a place to document and share your work</li>
<li>an index of practices searchable by location, medium, gender, allergy, discourse, and more</li>
<li>a federation of safe, user-moderated communities </li>
</ul>",
}
`;

exports[`gatsby-remark-embed-github-wiki inline username and repo overrides configuration 1`] = `
Object {
  "position": Position {
    "end": Object {
      "column": 49,
      "line": 1,
      "offset": 48,
    },
    "indent": Array [],
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "html",
  "value": "<p>Studio Visit App is:</p>
<ul>
<li>a slick but informal scheduling tool</li>
<li>a place to document and share your work</li>
<li>an index of practices searchable by location, medium, gender, allergy, discourse, and more</li>
<li>a federation of safe, user-moderated communities </li>
</ul>",
}
`;
